### ISSUE TITLE
Sort fails for patterns in non-long modes

### ISSUE THREAD

**Issue by initrd** [ogham/exa#496]
`exa -l --sort=size */file` works fine for files inside directories. However, `exa -1 --sort=size */file` or just `exa --sort=size */file` does not sort the output.

Five files of varying sizes inside directories:
```
$ exa -l */file
.rw-r--r-- 4.2M ashish 30 Apr 10:40 dir1/file
.rw-r--r-- 2.1M ashish 30 Apr 10:40 dir2/file
.rw-r--r-- 8.4M ashish 30 Apr 10:40 dir3/file
.rw-r--r-- 5.2M ashish 30 Apr 10:40 dir4/file
.rw-r--r--  10M ashish 30 Apr 10:40 dir5/file
```

Long listing, sorted by size (works as expected):
```
$ exa -l --sort=size */file
.rw-r--r-- 2.1M ashish 30 Apr 10:40 dir2/file
.rw-r--r-- 4.2M ashish 30 Apr 10:40 dir1/file
.rw-r--r-- 5.2M ashish 30 Apr 10:40 dir4/file
.rw-r--r-- 8.4M ashish 30 Apr 10:40 dir3/file
.rw-r--r--  10M ashish 30 Apr 10:40 dir5/file
```

Sorted by size, single column or without columns:
```
$ exa -1 --sort=size */file
dir1/file
dir2/file
dir3/file
dir4/file
dir5/file
$ exa --sort=size */file
dir1/file  dir2/file  dir3/file  dir4/file  dir5/file
```

Same with other sort options, here with `--sort=newest`:
```
$ exa -l --sort=newest */file
.rw-r--r-- 4.2M ashish 30 Apr 10:40 dir1/file
.rw-r--r-- 2.1M ashish 30 Apr 10:40 dir2/file
.rw-r--r--  10M ashish 30 Apr 10:40 dir5/file
.rw-r--r-- 8.4M ashish 30 Apr 10:40 dir3/file
.rw-r--r-- 5.2M ashish 30 Apr 10:40 dir4/file
$ exa -1 --sort=newest */file
dir1/file
dir2/file
dir3/file
dir4/file
dir5/file
$ exa --sort=newest */file
dir1/file  dir2/file  dir3/file  dir4/file  dir5/file
```

FWIW, `ls` handles as expected:
```
$ /bin/ls -l */file
-rw-r--r--  1 ashish  wheel   4194304 Apr 30 10:40 dir1/file
-rw-r--r--  1 ashish  wheel   2097152 Apr 30 10:40 dir2/file
-rw-r--r--  1 ashish  wheel   8388608 Apr 30 10:40 dir3/file
-rw-r--r--  1 ashish  wheel   5242880 Apr 30 10:40 dir4/file
-rw-r--r--  1 ashish  wheel  10485760 Apr 30 10:40 dir5/file
]$ /bin/ls -l -S */file
-rw-r--r--  1 ashish  wheel  10485760 Apr 30 10:40 dir5/file
-rw-r--r--  1 ashish  wheel   8388608 Apr 30 10:40 dir3/file
-rw-r--r--  1 ashish  wheel   5242880 Apr 30 10:40 dir4/file
-rw-r--r--  1 ashish  wheel   4194304 Apr 30 10:40 dir1/file
-rw-r--r--  1 ashish  wheel   2097152 Apr 30 10:40 dir2/file
$ /bin/ls -1 -S */file
dir5/file
dir3/file
dir4/file
dir1/file
dir2/file
$ /bin/ls -S */file
dir5/file       dir3/file       dir4/file       dir1/file       dir2/file
```

Edit: Running `exa v0.8.0` on macOS 10.14.4

**Comment by initrd** [comment]
After some more troubleshooting, I was able to narrow this down to the use of patterns and `--sort`:

```
$ exa -l --sort size
.rw-r--r-- 2.1M ashish 30 Apr 10:52 dir2-file
.rw-r--r-- 4.2M ashish 30 Apr 10:52 dir1-file
.rw-r--r-- 5.2M ashish 30 Apr 10:52 dir4-file
.rw-r--r-- 8.4M ashish 30 Apr 10:52 dir3-file
.rw-r--r--  10M ashish 30 Apr 10:52 dir5-file
$ exa -1 --sort size
dir2-file
dir1-file
dir4-file
dir3-file
dir5-file
$ exa --sort size
dir2-file  dir1-file  dir4-file  dir3-file  dir5-file
```
Add a pattern:
```
$ exa -l --sort=size dir[0-5]-file
.rw-r--r-- 2.1M ashish 30 Apr 10:52 dir2-file
.rw-r--r-- 4.2M ashish 30 Apr 10:52 dir1-file
.rw-r--r-- 5.2M ashish 30 Apr 10:52 dir4-file
.rw-r--r-- 8.4M ashish 30 Apr 10:52 dir3-file
.rw-r--r--  10M ashish 30 Apr 10:52 dir5-file
$ exa -1 --sort=size dir[0-5]-file
dir1-file
dir2-file
dir3-file
dir4-file
dir5-file
$ exa --sort=size dir[0-5]-file
dir1-file  dir2-file  dir3-file  dir4-file  dir5-file
```

**Comment by HarrisonMc555** [comment]
I just ran into this problem as well. I'm running v0.9.0 on macOS.

**Comment by ariasuni** [comment]
I can’t reproduce on latest exa 0.10.1. I believe this has been fixed by #756.

Note: when using shell globs, your shell change the glob into the list of correspond files, that is, running:

`exa -1 --sort=size file{1,2,3}`

is actually running:

`exa -1 --sort=size file1 file2 file3`
